magma/backend/mlir/hardware_module.py

```diff
@@ -498,8 +526,8 @@ def _visit(value, counter):
     def visit_coreir_primitive(self, module: ModuleWrapper) -> bool:
         inst = module.module
         defn = type(inst)
-        assert (defn.coreir_lib == "coreir" or defn.coreir_lib == "corebit")
-        if defn.coreir_name == "mem":
+        assert defn.coreir_lib in {"coreir", "corebit", "memory"}
+        if defn.coreir_name in {"mem", "sync_read_mem"}:
```

Comment:
prefer `or` clauses for fixed number of matches, as opposed to set inclusion. more readable and probably more performant to